tests/pkg_ucglib/main.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2018 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the Ucglib package.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#define TEST_OUTPUT_DUMMY 1
+#define TEST_OUTPUT_SDL 2
+#define TEST_OUTPUT_SPI 3
+
+#ifndef TEST_OUTPUT
+#error "TEST_OUTPUT not defined"
+#endif
+
+#if TEST_OUTPUT == TEST_OUTPUT_SPI
+#ifndef TEST_SPI
+#error "TEST_SPI not defined"
+#endif
+#ifndef TEST_DISPLAY
+#error "TEST_DISPLAY not defined"
+#endif
+#ifndef TEST_DISPLAY_EXT
+#error "TEST_DISPLAY_EXT not defined"
+#endif
+#ifndef TEST_PIN_CS
+#error "TEST_PIN_CS not defined"
+#endif
+#ifndef TEST_PIN_CD
+#error "TEST_PIN_CD not defined"
+#endif
+#ifndef TEST_PIN_RESET
+#error "TEST_PIN_RESET not defined"
+#endif
+#endif
+
+#include <stdio.h>
+
+#include "periph/gpio.h"
+
+#include "xtimer.h"
+#include "ucg.h"
+
+#include "logo.h"
+
+#if TEST_OUTPUT == TEST_OUTPUT_SPI
+/**
+ * @name    RIOT-OS pin maping of Ucglib pin numbers to RIOT-OS GPIO pins.
+ * @note    To minimize the overhead, you can implement an alternative for
+ *          ucg_com_riotos_hw_spi.
+ */
+static gpio_t pins[] = {
+    [UCG_PIN_CS] = TEST_PIN_CS,
+    [UCG_PIN_CD] = TEST_PIN_CD,
+    [UCG_PIN_RST] = TEST_PIN_RESET
+};
+
+/**
+ * @name   Bit mapping to indicate which pins are set.
+ */
+static uint32_t pins_enabled = (
+    (1 << UCG_PIN_CS) +
+    (1 << UCG_PIN_CD) +
+    (1 << UCG_PIN_RST)
+    );
+#endif
+
+int main(void)
+{
+    uint32_t screen = 0;
+    ucg_t ucg;
+
+#if TEST_OUTPUT == TEST_OUTPUT_DUMMY
+    /* initialize dummy output */
+    puts("Initializing dummy output.");
+
+    ucg_Init(&ucg, ucg_dev_dummy_cb, ucg_ext_none, NULL);
+#endif
+
+#if TEST_OUTPUT == TEST_OUTPUT_SDL
+    /* initialize to virtual SDL (native only) */
+    puts("Initializing to SDL.");
+
+    ucg_Init(&ucg, ucg_sdl_dev_cb, ucg_ext_none, NULL);
+#endif
+
+#if TEST_OUTPUT == TEST_OUTPUT_SPI
+    /* initialize to SPI */
+    puts("Initializing to SPI.");
+
+    ucg_SetPins(&ucg, pins, pins_enabled);
+    ucg_SetDevice(&ucg, TEST_SPI);
```

Comment:
I think it's better to use `SPI_DEV(TEST_SPI)`

Reply:
Fixed.